crates/rolldown_error/src/build_diagnostic/mod.rs

```diff
@@ -97,46 +94,80 @@ impl From<anyhow::Error> for BuildDiagnostic {
   }
 }
 
-#[derive(Debug, Default)]
-pub struct BatchedBuildDiagnostic(Vec<BuildDiagnostic>);
+#[derive(Debug)]
+pub enum BuildError {
+  Single(BuildDiagnostic),
+  Multi(Vec<BuildDiagnostic>),
```

Comment:
We tried this approach before and decide not to go this way. What's the motivation for this change?